libs/full/async_cuda/src/cuda_future.cpp

```diff
@@ -193,6 +193,7 @@ namespace hpx { namespace cuda { namespace experimental { namespace detail {
             if (status == cudaErrorNotReady)
             {
                 // this event has not been triggered yet
+                ++it;
```

Comment:
That is a very disturbing mistake, and could have been responsible for some dramatic slowdowns overall.